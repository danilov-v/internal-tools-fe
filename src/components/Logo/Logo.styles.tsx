import styled from 'styled-components';

type LogoWrapperProps = {
  size?: 'small' | 'default' | 'large';
  marginBottom?: string;
};

const getLogoSize = (size?: 'small' | 'default' | 'large'): string => {
  const sizeName = size || 'default';

  const sizeMap = {
    default: '3rem', // 48px
    small: '1.5rem', // 24px
    large: '8.5rem', // ~135px
  };

  return sizeMap[sizeName];
};

export const LogoWrapper = styled.div`
  height: ${({ size }: LogoWrapperProps) => getLogoSize(size)};
  width: ${({ size }: LogoWrapperProps) => getLogoSize(size)};
  width: 3rem;
  margin-bottom: ${(props: LogoWrapperProps) =>
    `${props.marginBottom || '0'}px`};
  display: block;
`;

export const LogoImage = styled.img`
  display: block;
  width: 100%;
  height: 100%;
`;
