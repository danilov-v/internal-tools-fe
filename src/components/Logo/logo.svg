<?xml version="1.0" encoding="UTF-8"?>
<!-- Created with Inkscape (http://www.inkscape.org/) -->
<svg width="100mm" height="100mm" version="1.1" viewBox="0 0 100 100" xmlns="http://www.w3.org/2000/svg" xmlns:cc="http://creativecommons.org/ns#" xmlns:dc="http://purl.org/dc/elements/1.1/" xmlns:rdf="http://www.w3.org/1999/02/22-rdf-syntax-ns#">
 <metadata>
  <rdf:RDF>
   <cc:Work rdf:about="">
    <dc:format>image/svg+xml</dc:format>
    <dc:type rdf:resource="http://purl.org/dc/dcmitype/StillImage"/>
    <dc:title/>
   </cc:Work>
  </rdf:RDF>
 </metadata>
 <g transform="translate(0,-197)">
  <g transform="matrix(.83575 0 0 .83575 2.3763 197)">
   <rect x="20.164" y="32.367" width="73.639" height="73.639" fill="#ccc"/>
   <g fill="#464646">
    <rect x="28.857" y="42.719" width="5.1759" height="55.288"/>
    <rect x="38.973" y="42.719" width="5.1759" height="55.288"/>
    <path d="m67.206 85.31-13.046-7.5323-1e-6 -15.065 13.046-7.5323 13.046 7.5323v15.065z"/>
   </g>
   <text x="18.719488" y="25.76049" fill="#cccccc" font-family="muller" font-size="16.991px" font-weight="500" letter-spacing="0px" stroke-width=".42478" word-spacing="0px" style="line-height:1.25" xml:space="preserve"><tspan x="18.719488" y="25.76049" fill="#cccccc" font-family="muller" font-weight="500" stroke-width=".42478">IT TOOLS</tspan></text>
  </g>
 </g>
</svg>
