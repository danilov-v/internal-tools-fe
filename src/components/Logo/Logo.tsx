import React from 'react';
import logoCaptioned from './logo_captioned.svg';
import logo from './logo.svg';
import * as S from './Logo.styles';

type LogoProps = {
  size?: 'default' | 'small' | 'large';
  marginBottom?: string;
  captioned?: boolean;
};

export const Logo: React.FC<LogoProps> = ({
  size,
  marginBottom,
  captioned,
}) => {
  return (
    <S.LogoWrapper size={size} marginBottom={marginBottom}>
      {captioned ? (
        <S.LogoImage src={logoCaptioned} alt="logo" />
      ) : (
        <S.LogoImage src={logo} alt="logo" />
      )}
    </S.LogoWrapper>
  );
};
