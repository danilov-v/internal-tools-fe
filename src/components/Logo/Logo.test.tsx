import React from 'react';
import { render } from '@testing-library/react';

import { Logo } from './Logo';

describe('<Logo />', () => {
  it('should render and match snapshot', () => {
    const component = render(<Logo />);

    expect(component.baseElement).toMatchSnapshot();
  });

  it('should render logo with image logo_captioned.svg', () => {
    const component = render(<Logo captioned />);
    const logoFileName = component.getByAltText('logo').getAttribute('src');

    expect(logoFileName).toEqual('logo_captioned.svg');
  });

  it('should render logo with image logo.svg', () => {
    const component = render(<Logo />);
    const logoFileName = component.getByAltText('logo').getAttribute('src');

    expect(logoFileName).toEqual('logo.svg');
  });

  it('should have widht and height 40px', () => {
    expect.assertions(2);

    const component = render(<Logo />);
    const logo = component.getByTitle('logo');

    expect(logo).toHaveStyle('height: 40px');
    expect(logo).toHaveStyle('width: 40px');
  });

  it('should have widht and height 20px with small size prop', () => {
    expect.assertions(2);

    const component = render(<Logo size="small" />);
    const logo = component.getByTitle('logo');

    expect(logo).toHaveStyle('height: 20px');
    expect(logo).toHaveStyle('width: 20px');
  });

  it('should have widht and height 135px with large size prop', () => {
    expect.assertions(2);

    const component = render(<Logo size="large" />);
    const logo = component.getByTitle('logo');

    expect(logo).toHaveStyle('height: 135px');
    expect(logo).toHaveStyle('width: 135px');
  });
});
