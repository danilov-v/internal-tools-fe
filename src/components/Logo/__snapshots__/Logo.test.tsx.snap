// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Logo /> should render and match snapshot 1`] = `
<body>
  <div>
    <div
      class="sc-AxjAm khWceK"
      title="logo"
    >
      <img
        alt="logo"
        class="sc-AxirZ dbNCkZ"
        src="logo.svg"
      />
    </div>
  </div>
</body>
`;
